File: market/src/main/java/com/wanted/market/global/auth/common/CustomAuthenticationEntryPoint.java
package com.wanted.market.global.auth.common;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.wanted.market.global.common.code.ResponseCode;
import com.wanted.market.global.common.response.BaseResponse;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.AuthenticationEntryPoint;
import org.springframework.stereotype.Component;

import java.io.IOException;

@Slf4j
@Component
public class CustomAuthenticationEntryPoint implements AuthenticationEntryPoint {

    @Override
    public void commence(HttpServletRequest request, HttpServletResponse response, AuthenticationException authException) throws IOException, ServletException {
        ObjectMapper objectMapper = new ObjectMapper();
        BaseResponse res = new BaseResponse(ResponseCode.NO_AUTHORIZATION);

        response.setContentType("application/json; charset=utf-8");
        response.setCharacterEncoding("utf-8");

        String responseValue = objectMapper.writeValueAsString(res);
        response.getWriter().write(responseValue);
    }
}

File: src/main/kotlin/wanted/market/api/common/security/CustomAuthenticationEntryPoint.kt
package wanted.market.api.common.security

import com.fasterxml.jackson.databind.ObjectMapper
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.http.HttpStatus.UNAUTHORIZED
import org.springframework.http.MediaType.APPLICATION_JSON_VALUE
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.AuthenticationEntryPoint
import org.springframework.stereotype.Component
import wanted.market.api.common.ErrorCode.HANDLE_ACCESS_DENIED

@Component
class CustomAuthenticationEntryPoint : AuthenticationEntryPoint {
    private val exceptionResponse: ErrorResponse = ErrorResponse.of(HANDLE_ACCESS_DENIED)

    override fun commence(httpServletRequest: HttpServletRequest, httpServletResponse: HttpServletResponse, e: AuthenticationException) {
        httpServletResponse.contentType = APPLICATION_JSON_VALUE
        httpServletResponse.status = UNAUTHORIZED.value()

        httpServletResponse.outputStream.use { os ->
            val objectMapper = ObjectMapper()
            objectMapper.writeValue(os, exceptionResponse)
            os.flush()
        }
    }
}